src/utils.jl

```diff
@@ -313,7 +314,11 @@ getline(ln) = Int(isexpr(ln, :line) ? ln.args[1] : ln.line)::Int
         CodeTracking.maybe_fixup_stdlib_path(path)
     end
 else
-    getfile(ln) = CodeTracking.maybe_fixup_stdlib_path(String(isexpr(ln, :line) ? ln.args[2] : ln.file)::String)
+    function getfile(ln::Union{LineTypes,Expr})
+        _getfile(ln::LineTypes) = ln.file::Symbol
+        _getfile(ln::Expr)      = ln.args[2]::Symbol # assuming ln.head === :line
+        return CodeTracking.maybe_fixup_stdlib_path(String(_getfile(ln)))
+    end
```

Comment:
Sorry for pushing such minor optimizations, these changes just circumvents inference limitations around https://github.com/JuliaLang/julia/issues/37342.

Reply:
Getting rid of invalidation risk is very worthwhile, no apologies needed!

My only concern is again the loss of validation that this is really an Expr of the expected type.